test/extended/authorization/per_node_update.go

```diff
@@ -0,0 +1,119 @@
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		err = pod.WaitForPodNameRunningInNamespace(ctx, oc.KubeClient(), actualPod.Name, actualPod.Namespace)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		actualPod, err = oc.KubeClient().CoreV1().Pods(oc.Namespace()).Get(ctx, actualPod.Name, metav1.GetOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		nodeScopedSAToken, err := exutil.ExecInPodWithResult(
+			oc.KubeClient().CoreV1(),
+			oc.UserConfig(),
+			actualPod.Namespace,
+			actualPod.Name,
+			"sleeper",
+			[]string{"cat", "/var/run/secrets/kubernetes.io/serviceaccount/token"},
+		)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		nodeScopedClientConfig := rest.AnonymousClientConfig(oc.UserConfig())
+		nodeScopedClientConfig.BearerToken = nodeScopedSAToken
+		nodeScopedClient, err := kubernetes.NewForConfig(nodeScopedClientConfig)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		saUser, err := nodeScopedClient.AuthenticationV1().SelfSubjectReviews().Create(ctx, &authenticationv1.SelfSubjectReview{}, metav1.CreateOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+		expectedUser := serviceaccount.MakeUsername(oc.Namespace(), "default")
+		o.Expect(saUser.Status.UserInfo.Username).To(o.Equal(expectedUser))
+		expectedNode := authenticationv1.ExtraValue([]string{actualPod.Spec.NodeName})
+		o.Expect(saUser.Status.UserInfo.Extra["authentication.kubernetes.io/node-name"]).To(o.Equal(expectedNode))
+
+		allowedConfigMap := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: oc.Namespace(),
+				Name:      actualPod.Spec.NodeName,
+			},
+		}
+		disallowedConfigMap := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: oc.Namespace(),
+				Name:      "unlikely-node",
+			},
+		}
+		actualAllowedConfigMap, err := nodeScopedClient.CoreV1().ConfigMaps(oc.Namespace()).Create(ctx, allowedConfigMap, metav1.CreateOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+		_, err = nodeScopedClient.CoreV1().ConfigMaps(oc.Namespace()).Create(ctx, disallowedConfigMap, metav1.CreateOptions{})
+		o.Expect(err).To(o.HaveOccurred())
+		o.Expect(err.Error()).To(o.ContainSubstring("this user may only modify configmaps that belong to the node the pod is running on"))
+
+		// now create so we can see the update cases
+		actualDisallowedConfigMap, err := oc.AdminKubeClient().CoreV1().ConfigMaps(oc.Namespace()).Create(ctx, disallowedConfigMap, metav1.CreateOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		actualAllowedConfigMap, err = nodeScopedClient.CoreV1().ConfigMaps(oc.Namespace()).Update(ctx, actualAllowedConfigMap, metav1.UpdateOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+		_, err = nodeScopedClient.CoreV1().ConfigMaps(oc.Namespace()).Update(ctx, actualDisallowedConfigMap, metav1.UpdateOptions{})
+		o.Expect(err).To(o.HaveOccurred())
+		o.Expect(err.Error()).To(o.ContainSubstring("this user may only modify configmaps that belong to the node the pod is running on"))
```

Comment:
This is the meat.  The restricted serviceaccount token can only modify its own node's configmap, but other users are still unconstrained.